flexget/plugins/metainfo/tvmaze_lookup.py

```diff
@@ -0,0 +1,171 @@
+        'tvmaze_series_rating': 'rating',
+        'tvmaze_series_episodes': lambda show: [episodes.title for episodes in show.episodes]
+    }
+
+    # Episode info
+    episode_map = {
+        'tvmaze_episode_name': 'title',
+        'tvmaze_season': 'season_number',
+        'tvmaze_episode': 'number',
+        'tvmaze_episode_id': 'tvmaze_id',
+        'tvmaze_episode_airdate': 'airdate',
+        'tvmaze_episode_url': 'url',
+        'tvmaze_episode_original_image': 'original_image',
+        'tvmaze_episode_medium_image': 'medium_image',
+        'tvmaze_episode_airstamp': 'airstamp',
+        'tvmaze_episode_runtime': 'runtime'
+
+    }
+
+    schema = {'type': 'boolean'}
+
+    def lazy_series_lookup(self, entry):
+        """Does the lookup for this entry and populates the entry fields."""
+        with Session() as session:
+            lookupargs = {'title': entry.get('series_name', eval_lazy=False),
+                          'year': entry.get('year', eval_lazy=False),
+                          'tvmaze_id': entry.get('tvmaze_id', eval_lazy=False),
+                          'tvdb_id': entry.get('tvdb_id', eval_lazy=False),
+                          'tvrage_id': entry.get('tvrage_idk', eval_lazy=False),
+                          'session': session}
+            try:
+                series = series_lookup(**lookupargs)
+            except LookupError as e:
+                log.debug(e.args[0])
+            else:
+                entry.update_using_map(self.series_map, series)
+        return entry
+
+    def lazy_episode_lookup(self, entry):
+        with Session(expire_on_commit=False) as session:
+            lookupargs = {'title': entry.get('series_name', eval_lazy=False),
+                          'year': entry.get('year', eval_lazy=False),
+                          'tvmaze_id': entry.get('tvmaze_id', eval_lazy=False),
+                          'tvdb_id': entry.get('tvdb_id', eval_lazy=False),
+                          'tvrage_id': entry.get('tvrage_id', eval_lazy=False),
+                          'series_season': entry.get('series_season', eval_lazy=False),
+                          'series_episode': entry.get('series_episode', eval_lazy=False),
+                          'series_date': entry.get('series_date', eval_lazy=False),
+                          'series_id_type': entry.get('series_id_type', eval_lazy=False),
+                          'session': session}
+            try:
+                episode = episode_lookup(**lookupargs)
+            except LookupError as e:
+                log.debug(e.args[0])
+            else:
+                entry.update_using_map(self.episode_map, episode)
+        return entry
+
+    # Run after series and metainfo series
+    @plugin.priority(110)
+    def on_task_metainfo(self, task, config):
+        if not config:
+            return
+
+        for entry in task.entries:
+            if entry.get('series_name') or entry.get('tvdb_id', eval_lazy=False):
```

Comment:
Yeah,  tvrage too. Forgot to add those. Also I'll add some tests with series plugin using set.